spec/unleash/constraint_spec.rb

```diff
@@ -384,23 +388,38 @@
       expect(constraint.matches_context?(context)).to be false
     end
 
-    it 'rejects constraint construction for invalid value types for operator' do
+    it 'gracefully handles invalid constraint operators' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', 'dev', inverted: true)
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', ['dev'], inverted: true)
+      expect(constraint.matches_context?(context)).to be true
+    end
+
+    it 'warns about constraint construction for invalid value types for operator' do
       array_constraints = ['STR_CONTAINS', 'STR_ENDS_WITH', 'STR_STARTS_WITH', 'IN', 'NOT_IN']
 
       array_constraints.each do |operator_name|
         Unleash::Constraint.new('env', operator_name, [])
-        expect do
-          Unleash::Constraint.new('env', operator_name, '')
-        end.to raise_error
+        expect(Unleash.logger).to receive(:warn).with("value is a String, operator is expecting an Array")
+        Unleash::Constraint.new('env', operator_name, '')
       end
 
       string_constraints = ['NUM_EQ', 'NUM_GT', 'NUM_GTE', 'NUM_LT', 'NUM_LTE',
                             'DATE_AFTER', 'DATE_BEFORE', 'SEMVER_EQ', 'SEMVER_GT', 'SEMVER_LT']
       string_constraints.each do |operator_name|
         Unleash::Constraint.new('env', operator_name, '')
```

Comment:
No need to test that the creation of the object succeeded w/o an exception anymore. It now should be a safe assumption.
```suggestion
```

Reply:
I added this test based on review you had earlier. Its not testing the object creation, its checking that unleash logs a warn message.

Reply:
We have two `Unleash::Constraint.new` in this test (per `operator_name`). We only need to test the "sad" path, that the logger is getting called. No need to test the happy path....

That is at least what I meant to say.